src/graph/FetchVerticesExecutor.cpp

```diff
@@ -93,35 +48,272 @@ Status FetchVerticesExecutor::prepareVids() {
+    }
+
+    if (tagNames.size() == 1 && *tagNames[0] == "*") {
+        auto tagsStatus = ectx()->schemaManager()->getAllTag(spaceId_);
+        if (!tagsStatus.ok()) {
+            return tagsStatus.status();
+        }
+        for (auto& tagName : std::move(tagsStatus).value()) {
+            auto tagIdStatus = ectx()->schemaManager()->toTagID(spaceId_, tagName);
+            if (!tagIdStatus.ok()) {
+                return tagIdStatus.status();
+            }
+            auto tagId = tagIdStatus.value();
+            tagNames_.push_back(tagName);
+            tagIds_.push_back(tagId);
+            auto result = tagNameSet_.emplace(tagName);
+            if (!result.second) {
+                return Status::Error(folly::sformat("tag({}) was dup", tagName));
+            }
+        }
+    } else {
+        for (auto tagName : tagNames) {
+            auto tagStatus = ectx()->schemaManager()->toTagID(spaceId_, *tagName);
+            if (!tagStatus.ok()) {
+                return tagStatus.status();
+            }
+            auto tagId = tagStatus.value();
+            tagNames_.push_back(*tagName);
+            tagIds_.push_back(tagId);
+            auto result = tagNameSet_.emplace(*tagName);
+            if (!result.second) {
+                return Status::Error(folly::sformat("tag({}) was dup", *tagName));
+            }
+        }
+    }
+
+    return status;
+}
+
+Status FetchVerticesExecutor::prepareYield() {
+    colNames_.emplace_back("VertexID");
+    colTypes_.emplace_back(nebula::cpp2::SupportedType::VID);
+    if (yieldClause_ == nullptr) {
+        // determine which columns to return after received response from storage.
+        for (unsigned i = 0; i < tagNames_.size(); i++) {
+            auto& tagName = tagNames_[i];
+            auto tagId = tagIds_[i];
+            std::shared_ptr<const meta::SchemaProviderIf> tagSchema =
+                ectx()->schemaManager()->getTagSchema(spaceId_, tagId);
+            if (tagSchema == nullptr) {
+                return Status::Error("No tag schema for %s", tagName.c_str());
+            }
+            for (auto iter = tagSchema->begin(); iter != tagSchema->end(); ++iter) {
+                auto *prop = iter->getName();
+                storage::cpp2::PropDef pd;
+                pd.owner = storage::cpp2::PropOwner::SOURCE;
+                pd.name = prop;
+                pd.id.set_tag_id(tagId);
+                props_.emplace_back(std::move(pd));
+            }
+        }
+    } else {
+        for (auto *col : yieldClause_->columns()) {
+            if (!col->getFunName().empty()) {
+                return Status::SyntaxError("Do not support aggregated query with fetch prop on.");
+            }
+
+            if (col->expr()->isInputExpression()) {
+                auto *inputExpr = dynamic_cast<InputPropertyExpression*>(col->expr());
+                auto *colName = inputExpr->prop();
+                if (*colName == "*") {
```

Comment:
Excellent.  Could we combine the logic about "*" both in InputPropertyExpression and VariablePropertyExpression into one?

Reply:
may have trouble due to 

```
Expression *expr = new InputPropertyExpression(new std::string(prop));
Expression *expr = new VariablePropertyExpression(alias, new std::string(prop));
```